prov/efa/src/efa_fabric.c

```diff
@@ -536,13 +536,20 @@ static int efa_get_device_attrs(struct efa_context *ctx, struct fi_info *info)
 	info->domain_attr->resource_mgmt	= FI_RM_DISABLED;
 	info->domain_attr->mr_cnt		= base_attr->max_mr;
 
-#if HAVE_LIBCUDA
-	if (info->ep_attr->type == FI_EP_RDM &&
-	    efa_get_gdr_support(ctx->ibv_ctx->device->name) == 1) {
-		info->caps			|= FI_HMEM;
-		info->tx_attr->caps		|= FI_HMEM;
-		info->rx_attr->caps		|= FI_HMEM;
-		info->domain_attr->mr_mode	|= FI_MR_HMEM;
+#ifdef HAVE_LIBCUDA
+	if (info->ep_attr->type == FI_EP_RDM) {
+		gdr_support = efa_get_gdr_support(ctx->ibv_ctx->device->name);
+		if (gdr_support !=0 && gdr_support !=1) {
```

Comment:
This feels like the wrong fix to me.  efa_get_gdr_support() should have well defined return values.  It feels like -1, 0, 1 are the valid values, but I'll leave that to you.  The efa_get_gdr_support() function itself should handle coercing the return value into the specified options.

Reply:
Good point! In new revision, I moved this part to inside ```efa_get_gdr_support```, and added an assertion here.